src/test/java/picard/sam/BamErrorMetric/BaseErrorCalculationTest.java

```diff
@@ -0,0 +1,153 @@
+        final int length = refBases.length;
+
+        for (int i = 0; i < length; i++) {
+
+            SamLocusIterator.LocusInfo locusInfo = new SamLocusIterator.LocusInfo(samSequenceRecord, i + 1);
+            SamLocusIterator.RecordAndOffset recordAndOffset1 = new SamLocusIterator.RecordAndOffset(samRecord1, i);
+            SamLocusIterator.RecordAndOffset recordAndOffset2 = new SamLocusIterator.RecordAndOffset(samRecord2, i);
+            locusInfo.add(recordAndOffset1);
+            locusInfo.add(recordAndOffset2);
+
+            final SAMLocusAndReference locusAndReference = new SAMLocusAndReference(locusInfo, refBases[i]);
+
+            overlappingErrorCalculator1.addBase(recordAndOffset1, locusAndReference);
+            overlappingErrorCalculator2.addBase(recordAndOffset2, locusAndReference);
+        }
+
+        final OverlappingErrorMetric metric1 = overlappingErrorCalculator1.getMetric();
+        metric1.calculateDerivedFields();
+        Assert.assertEquals(metric1.TOTAL_BASES, length);
+        Assert.assertEquals(metric1.NUM_BASES_WITH_OVERLAPPING_READS, length);
+        Assert.assertEquals(metric1.NUM_DISAGREES_WITH_REFERENCE_ONLY, 2L);
+        Assert.assertEquals(metric1.NUM_DISAGREES_WITH_REF_AND_MATE, 1L);
+        Assert.assertEquals(metric1.NUM_THREE_WAYS_DISAGREEMENT, 1L);
+
+        final OverlappingErrorMetric metric2 = overlappingErrorCalculator1.getMetric();
+        metric2.calculateDerivedFields();
+        Assert.assertEquals(metric2.TOTAL_BASES, length);
+        Assert.assertEquals(metric2.NUM_BASES_WITH_OVERLAPPING_READS, length);
+        Assert.assertEquals(metric2.NUM_DISAGREES_WITH_REFERENCE_ONLY, 2L);
+        Assert.assertEquals(metric2.NUM_DISAGREES_WITH_REF_AND_MATE, 1L);
+        Assert.assertEquals(metric2.NUM_THREE_WAYS_DISAGREEMENT, 1L);
+    }
+
+    @DataProvider( )
+    public Object[][] testOverlappingErrorCalculatorWithManyReadsData() throws IOException {
+        final File temp = File.createTempFile("Overlapping", ".bam");
+        temp.deleteOnExit();
+
+        try (
+                final ReferenceSequenceFileWalker referenceSequenceFileWalker =
+                        new ReferenceSequenceFileWalker(new File("testdata/picard/sam/BamErrorMetrics/chrM.reference.fasta"))) {
+
+            final SAMRecordSetBuilder builder = new SAMRecordSetBuilder();
+            builder.getHeader().setSequenceDictionary(referenceSequenceFileWalker.getSequenceDictionary());
+
+            for (int i = 0; i < 4000; i++) {
+                builder.addPair("Read" + String.valueOf(i), 0, 1, 1,
+                        false, false, "36M", "36M", true, false, 20);
+            }
+
+            try (final SAMFileWriter writer = new SAMFileWriterFactory()
+                    .makeBAMWriter(builder.getHeader(), false, temp)) {
+                builder.forEach(writer::addAlignment);
+            }
+        }
+        return new Object[][]{{temp}};
```

Comment:
Doesn't seem to be a benefit to having a dataprovider that has a single return value and only one use. Making this a method like `createTestDataFileForXXXX()` and having it return `File` might be clearer.

Reply:
I wanted to separate the time for creation of the file from the time or the test....